fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeStmt.java

```diff
@@ -122,12 +125,40 @@ public List<Table> getTables() throws AnalysisException {
         return tables;
     }
 
+    public List<String> getPartitionNames() {
+        if (partitionNames == null) {
+            return Lists.newArrayList();
+        }
+        return partitionNames.getPartitionNames();
+    }
+
+    public Map<Long, List<String>> getTableIdToPartitionName() throws AnalysisException {
+        Preconditions.checkArgument(isAnalyzed(),
+                "The partitionIds must be obtained after the parsing is complete");
+        Map<Long, List<String>> tableIdToPartitionName = Maps.newHashMap();
+
+        for (Table table : getTables()) {
+            table.readLock();
+            try {
+                OlapTable olapTable = (OlapTable) table;
+                List<String> partitionNames = getPartitionNames();
```

Comment:
If you allow users to specify partitions and allow users to specify multiple tables at the same time, the user-specified partitions and tables must correspond to each other. So you need a map structure to store related information, not a unified partitionnames

Reply:
Now it supports analysing all tables under the a database; supports analysing a single table (user can specify columns, specify partitions). In the case of specified conditions, it is not supported to analyze multiple tables with one statement. If this is necessary, we can discuss and support this scenario.